webapi/src/main/java/org/sakaiproject/webapi/beans/RosterRestBean.java

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ * Copyright 2015 sakaiproject.org Licensed under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/ECL-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+package org.sakaiproject.webapi.beans;
+
+import lombok.Data;
+import lombok.NonNull;
+
+@Data
+public class RosterRestBean {
+
+    private String id;
+    private String title;
+    private RosterType type;
+
+    public static enum RosterType {
+        CMPROVIDED, CMREQUESTED, MANREQUESTED
+    }
+
+    public RosterRestBean(@NonNull String id, @NonNull RosterType type) {
```

Comment:
These @NonNull annotations confuse me a bit. Why not test the vars for null and throw IllegalArgumentException if they are empty? Does NonNull account for blank strings like StringUtils? From the lombok docs it seems that it throws a NPE. But, if you don't add the annotation and try to use the var, you get an NPE anyway? They seem pointless to me.